apps-rendering/src/components/Standfirst/GalleryStandfirst.tsx

```diff
@@ -24,16 +27,60 @@ const styles = css`
 	${from.desktop} {
 		${grid.span('centre-column-start', 6)}
 	}
+
+	${darkModeCss`
+		color: ${neutral[86]};
+	`}
+`;
+
+const bylineStyles = css`
+	padding-top: ${remSpace[2]};
 `;
 
+const renderContent = (
+	standfirst: DocumentFragment,
+	format: ArticleFormat,
+	byline: string,
+	bylineHtml: Option<DocumentFragment>,
+): ReactNode => {
+	const rendered = renderStandfirstText(standfirst, format);
+	const bylineInStandfirst =
+		byline !== '' && standfirst.textContent?.includes(byline);
+
+	// Immersives append the byline to the standfirst.
+	// Sometimes CAPI includes this within the standfirst HTML,
+	// sometimes we have to add it ourselves from the byline HTML.
+	if (!bylineInStandfirst && bylineHtml.kind === OptionKind.Some) {
+		return (
+			<>
+				{rendered}
+				<address css={bylineStyles}>
+					<p>by {renderStandfirstText(bylineHtml.value, format)}</p>
+				</address>
+			</>
+		);
+	}
+
+	return rendered;
+};
```

Comment:
They shouldn’t (I know of no examples). And the example given should be sent to Central Prod as I can’t see a reason why it’s been misedited such! (there were some genuine reasons where the byline was used in standfirst as a part of a sentence, but not here).

Galleries have their own silly quirks:
1. Sometimes extensive time-sensitive info is written [into first](https://www.theguardian.com/artanddesign/gallery/2020/sep/16/the-other-side-of-rene-burri-swiss-photographer-in-pictures) (or last) caption, because Editorial feels it would be too big as part of standfirst.
2. Sometimes, All photographs by, instead of being written into Byline (like [here](https://www.theguardian.com/stage/gallery/2020/sep/16/from-top-girls-to-the-skriker-the-best-of-caryl-churchill-in-pictures-tristram-kenton)), is written [into first](https://www.theguardian.com/artanddesign/gallery/2020/sep/16/the-other-side-of-rene-burri-swiss-photographer-in-pictures), [or last](https://www.theguardian.com/artanddesign/gallery/2020/oct/10/mesmerising-urban-photography-in-pictures), image caption (no idea what’s the reasoning, tried to fight it, lost).
3. Editorial expressed a need for a footer for when a legal change had to be communicated (just once).
4. Dotcom is doing weird stuff enhancing first emboldened para into an H2 (because Composer Gallery editor doesn’t allow H2, ma do in the future).

Some of the above are Editorial (bad) habits. But platforms and tools (pt. 4) can maybe help alleviate just some of these…

Reply:
Ah ok I think I see what's happening here. So I've made it more complex that it needs to be by using the immersive work as a base. I'm thinking I can just separate this out and add a `<GalleryByline />` component. Sound good?

Reply:
> I'm thinking I can just separate this out and add a `<GalleryByline />` component. Sound good?

Sounds good, it seems we don't need to worry about this problem for galleries.

@paperboyo couple of questions on (3) and (4). On 3), what sort of footer? Do you have an example piece we could look at to make sure this works? On (4) are we talking about paragraphs within captions? AR has logic to do this too, although I wonder if a `<strong>` (which is what CAPI gives us) might be more appropriate?

Reply:
Ad. 3. Sadly, can’t find an example. Will fwd you the only request that was made about it. It didn’t go anywhere, but the need may be there. Something to ponder only, really.
Ad. 4. Yes, emboldened (first) paras within captions. One might think, Composer should add H2s to image captions. But practically, I’m not sure: it would need to be restricted to one, opening H2 (sic!) and would require massive habit change for Editorial. Maybe we should just live with the silly cleaner. Although it consistently barfs for when something else is emboldened further down or copy/paste fails, haha, see [here](https://www.theguardian.com/news/gallery/2022/aug/24/ukraine-independence-day-and-reading-festival-wednesdays-best-photos#img-11)

![image](https://user-images.githubusercontent.com/6032869/186912038-e4e4d8a8-9774-4d5d-9599-a470070c11e4.png)

CAPI:
![image](https://user-images.githubusercontent.com/6032869/186912353-ec73fd24-fc8a-4a78-8069-ed798759863b.png)